### PR TITLE
Tag - migrate to stylable

### PR DESCRIPTION
**Pull request by Zivilekin**

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6432

File: src/Tag/Tag.st.css [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+  background-color: value(B50);
+  border-color: value(B30);
+}
+
+.root:disabled {
+  background-color: value(D10-05);
+  color: value(D10-30);
+}
+
+:global(.rtl) .root:size(tiny),
```
Comment (EgoziE) [inline src/Tag/Tag.st.css:208]: I see all our `:global(.rtl)` styles here are related to margin and padding.
Let's remove **all** of them in favor of:
- margin-block-start
- margin-block-end
- padding-block-start
- padding-block-end

Note, don't use `margin-block` or `padding-block` without the starts/end suffix

**Review by EgoziE — CHANGES_REQUESTED** [review]

File: src/Tag/Tag.js [hunk 2]
```diff
@@ -68,39 +72,45 @@ class Tag extends React.PureComponent {
...
-        [styles.clickable]: onClick !== noop,
-        [styles.hoverable]: !disabled && onClick !== noop,
-      },
-    );
-  }
-
-  render() {
-    const { id, onClick, maxWidth, dataHook } = this.props;
-
+    const clickable = onClick !== noop;
```
Comment (mykas) [inline src/Tag/Tag.js:88]: Lets remove this noop  usage and instead do: `!!onClick`

File: src/Tag/Tag.st.css [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+.root {
+  -st-states: withRemoveButton, withThumb, disabled, clickable, hoverable,
+    size(enum(tiny, small, medium, large)),
+    theme(enum(standard, error, warning, dark, neutral, light, lightOutlined));
+  box-sizing: border-box;
+  position: relative;
+  display: inline-flex;
+  align-items: center;
+  transition: background-color ease 300ms;
+  max-width: -moz-available;
```
Comment (mykas) [inline src/Tag/Tag.st.css:16]: I think that browser based prefixes are handled by the webpack already. No need to add special handling

File: src/Tag/Tag.st.css [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  max-width: -webkit-fill-available;
+  max-width: stretch;
```
Comment (mykas) [inline src/Tag/Tag.st.css:18]: I think we have a problem here. `stretch` is not supported by most of the browsers like safari, firefox: https://developer.mozilla.org/en-US/docs/Web/CSS/max-width

Some kind of other solution must be found here

File: src/Tag/Tag.st.css [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+}
+.root:size(large) .removeButton {
+  margin-inline-start: 12px;
+}
+.root:not(:disabled):clickable {
+  cursor: pointer;
+}
+
+/* Themes */
+.root:not(:disabled):theme(standard) {
```
Comment (mykas) [inline src/Tag/Tag.st.css:151]: I wonder if you remove this `:not(:disabled)`. Does it work?

File: src/Tag/Tag.st.css (hunk 4 above)
Comment (mykas) [reply src/Tag/Tag.st.css:16]: to resolve the issue a separate task has been created.

**Review by mykas — APPROVED** [review]
Good job, Zivile! Really, I mean it!

**Pull request merged**